integration_tests/src/main/python/string_test.py

```diff
@@ -73,6 +73,13 @@ def test_split_positive_limit():
             'split(a, "C", 3)',
             'split(a, "_", 999)'))
 
+def test_split_escaped_dot():
+    data_gen = mk_str_gen(r'([0-9]\.){1,4}')
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark : unary_op_df(spark, data_gen).selectExpr(
+            'split(a, "[\\\\.]", 2)'
```

Comment:
Can you add a comment how many escapes are you demonstrating?

In Python

```Python
>>> 'split(a, "[\\\\.]", 2)' == r'split(a, "[\\.]", 2)'
True
```

so presumably we pass `split(a, "[\\.]", 2)` to the SQL parser. And `"[\\.]"`, `"[\.]`, `"[.]"` all produce  the same result
```sql
spark-sql (default)> SELECT SPLIT(a, "[\\.]", 2) FROM VALUES '1245.xyz', 'abdd.erg' AS DATA(a);
["1245","xyz"]
["abdd","erg"]
Time taken: 0.011 seconds, Fetched 2 row(s)
spark-sql (default)> SELECT SPLIT(a, "[\.]", 2) FROM VALUES '1245.xyz', 'abdd.erg' AS DATA(a);
["1245","xyz"]
["abdd","erg"]
Time taken: 0.011 seconds, Fetched 2 row(s)
spark-sql (default)> SELECT SPLIT(a, "[.]", 2) FROM VALUES '1245.xyz', 'abdd.erg' AS DATA(a);
["1245","xyz"]
["abdd","erg"]
Time taken: 0.011 seconds, Fetched 2 row(s)
```

Reply:
We have to double-escape when passing the strings from python to jvm, so python `\\\\` becomes java `\\`. This applies to many of the tests in regexp_test.

In the first query you pasted here `[\\.]` would match either `\` or `.` and `[\.]` and `[.]` would only match `.`. All would produce the expected results in this case.

Reply:
Just to demonstrate, here is the first query with the input changed to use `\` instead of `.`:

```
spark-sql> SELECT SPLIT(a, "[\\.]", 2) FROM VALUES '1245\\xyz', 'abdd\\erg' AS DATA(a);
["1245\xyz"]
["abdd\erg"]
```

Reply:
IMO it's more clear if we have a comment, here is

1) an actual SQL regex being tested 
2) here are the hoops we are jumping from Python to get there.

and that raw strings might be more intuitive

```Python
>>> df.selectExpr(r'SPLIT(a, "[\\\\.]", 2)').explain()
== Physical Plan ==
*(1) Project [split(a#2, [\\.], 2) AS split(a, [\\.], 2)#67]
+- *(1) Scan ExistingRDD[a#2]


>>> df.selectExpr('SPLIT(a, "[\\\\.]", 2)').explain()
== Physical Plan ==
*(1) Project [split(a#2, [\.], 2) AS split(a, [\.], 2)#69]
+- *(1) Scan ExistingRDD[a#2]
```

Reply:
I updated the test to use raw strings and added a comment. Let me know what you think, and I can apply the comment to other tests in a follow-on PR.

Reply:
LGTM

My angle was not to use JVM as a reference but the string inside the Spark plan. And raw strings are the most compact way to get there via different APIs. And looks like SQL has raw strings as well

Spark Plan: `split(a#235, [\\.], 2)`

SQL:

 ```sql
SELECT split(a, r"[\\.]", 2) FROM df;
```

Python:

```Python

df.selectExpr(r'split(a, "[\\\\.]", 2)')

df.selectExpr(r'split(a, r"[\\.]", 2)')

import pyspark.sql.functions as f
df.select(f.split("a", r"[\\.]", 2))
```

Scala:

```Scala
df.selectExpr("""split(a, r"[\\.]" ,2)""")

import org.apache.spark.sql.{functions => f}
df.select(f.split($"a", raw"[\\.]", 2))
```